File: Easy/Python/2022. Convert 1D Array Into 2D Array.py
class Solution(object):
    def construct2DArray(self, original, m, n):
        if len(original) != m * n:
            return []
        
        arr = [[0] * n for _ in range(m)]
        row = 0
        col = 0

        for num in original:
            arr[row][col] = num
            col += 1
            if col == n:
                col = 0
                row += 1

        return arr

File: Easy/Java/2022. Convert 1D Array Into 2D Array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        if(original.length != m * n)
            return new int[0][];
        
        int[][] arr = new int[m][n];
        int row = 0;
        int col = 0;

        for(int num : original){
            arr[row][col] = num;
            col++;
            if(col == n){
                col = 0;
                row++;
            }
        }

        return arr;
    }
}
